scipy/optimize/_shgo.py

```diff
@@ -200,7 +200,49 @@ def shgo(
         Feedback:
 
         * disp : bool (L)
-            Set to True to print convergence messages.
+            Set to True to print convergence messages. This setting applies
+            mainly to compiled subroutines used during optimization. The
+            native Python code is mainly set up to log messages using the
```

Comment:
This is too much information for a docstring, and all of the logging discussion is irrelevant to `disp`. Elsewhere in the docstring (probably `Notes`), just mention that the Python parts use the standard `logging` framework and log to `scipy.optimize.shgo`. Do not include demos of how to set up logging.

Reply:
Thank you, I agree.